src/test/ui/rfc-2497-if-let-chains/disallowed-positions.rs

```diff
@@ -218,18 +218,21 @@ fn inside_const_generic_arguments() {
         true && let 1 = 1 //~ ERROR `let` expressions are not supported here
         //~^ ERROR constant contains unimplemented expression type
         //~| ERROR constant contains unimplemented expression type
+        //~| ERROR `match` is not allowed in a `const`
```

Comment:
@centril, the errors from the HIR const-checker are not as descriptive as the current one. Do you think we should try to suppress the error here?

Reply:
I'd say it's not worth the complexity because this case is fairly pathological; unlikely to be seen by users?